src/components/Post/PostDetailComponent.jsx

```diff
@@ -1,16 +1,127 @@
 import React, { Component } from 'react';
-import { Segment } from 'semantic-ui-react';
+import { Segment, Image } from 'semantic-ui-react';
+import { withRouter, Link } from 'react-router-dom';
+import GoogleMapReact from 'google-map-react';
+
+import MapMarker from '../Home/MapMarkers';
 
 class NewPost extends Component {
+  state = {
+    zoom:15,
+  }
+  
+  postColorHandler = (post) => {
+    let postColor = '';
+    if (post.user_rating === 'Gnarly') {
+      postColor = 'red';
+    } else if (post.user_rating === 'Good') {
+      postColor = 'orange';
+    } else if (post.user_rating === 'Fair') {
+      postColor = 'yellow';
+    } else if (post.user_rating === 'Poor') {
+      postColor = 'teal';
+    } else if (post.user_rating === 'Flat') {
+      postColor = 'grey';
+    }
+    return postColor;
```

Comment:
You might want to use switch case statement here. So I didn't know this until later but switch case statements are considered to compile faster than using `else if...` conditionals.

REFERENCE: https://www.quora.com/How-is-a-switch-case-faster-than-if-else-block

Reply:
ah i see makes sense
and good to know